spark/src/main/scala/org/apache/spark/sql/delta/ConflictChecker.scala

```diff
@@ -547,6 +548,48 @@ private[delta] class ConflictChecker(
     currentTransactionInfo = currentTransactionInfo.copy(actions = newActions)
   }
 
+  /** Ensure that commitInfo.commitTimestamp is monotonically increasing. */
+  private def resolveTimestampOrderingConflicts(): Unit = {
+    val isICTCurrentlyEnabled =
+      DeltaConfigs.IN_COMMIT_TIMESTAMPS_ENABLED.fromMetaData(currentTransactionInfo.metadata)
+    if (!isICTCurrentlyEnabled) {
```

Comment:
This was intentional so that I don't need another comment around L577 explaining that if we are at that point, then we already know that ICT is enabled in the current txn. It can be removed if you find it unnecessary from that perspective.

Reply:
In this case, the whole function is gated on ICT being enabled in the current transaction. 
We shouldn't need a lot of reminders in the function body?

Meanwhile, the danger of redundant control flow is high complexity: every check we make implies a branch point in the logic, and somebody has to think about what happens on both sides of the branch. If one side is impossible, we risk wasting a lot of effort reasoning about (and even coding up) the dead case.